feat: support date types for range components (#261)
appbaseio/Docs: content/docs/reactivesearch/v3/range/rangeslider.md

|       `basic_time_no_millis` |                                                                         `HHmmssZ`                                                                          |
|                       `date` |                                                                        `yyyy-MM-dd`                                                                        |
|                 `basic_date` |                                                                         `yyyyMMdd`                                                                         |
|            `basic_date_time` |                                                                  `yyyyMMdd'T'HHmmss.SSSZ`                                                                  |
|  `basic_date_time_no_millis` |                                                                    `yyyyMMdd'T'HHmmssZ`                                                                    |
|        `date_time_no_millis` |                                                                 `yyyy-MM-dd'T'HH:mm:ssZZ`      

> Note: `queryFormat` is mandatory to pass when dealing with date types.
-   **calendarInterval** `String` [optional]

>     sets the histogram bar interval, applicable dealing with date-types. Defaults to `month`. It can accept one of the following: `year`, `quarter`, `month`, `week`, `day`, `hour`, and `minute`. 

Reviewer: ```suggestion
    sets the histogram bar interval, applicable when the dataField set is of type `date`. Default value is set to `month`. It can accept one of the following: `year`, `quarter`, `month`, `week`, `day`, `hour`, and `minute`. 
```

Reviewer: @mohdashraf010897 The default value for `calendarInterval` should be auto-set by the library using the start and end values. We can update the docs for the same.
